File: grind75/go/valid-parentheses.go
package main

import (
	"fmt"
	"strings"
)

func main() {

	result := isValid("()[]{}")

	fmt.Println(result)
}

func isValid(s string) bool {
	if len(s)%2 != 0 {
		return false
	}

	for strings.Contains(s, "()") || strings.Contains(s, "[]") || strings.Contains(s, "{}") {
		s = strings.ReplaceAll(s, "()", "")
		s = strings.ReplaceAll(s, "[]", "")
		s = strings.ReplaceAll(s, "{}", "")
	}

	return len(s) == 0
}

File: grind75/rust/valid-parentheses.rs
fn main() {
    pub fn is_valid(mut s: String) -> bool {
        if s.len() % 2 != 0 {
            return false;
        }

        while s.contains("()") || s.contains("{}") || s.contains("[]") {
            s = s.replace("()", "");
            s = s.replace("[]", "");
            s = s.replace("{}", "");
        }

        return s.len() == 0;
    }

    let result = is_valid("[](){}".to_owned());

    println!("{:?}", result);
}
